components/match2/wikis/brawlstars/match_legacy.lua

```diff
@@ -0,0 +1,203 @@
+	for key, item in pairs(streams) do
+		table.insert(
+			subObjectTable,
+			'Has match ' .. key .. '=' .. item
+		)
+	end
+
+	for key, map in pairs(match2.match2games) do
+		table.insert(
+			subObjectTable,
+			'Has match vodgame' .. key .. '=' .. (map.vod or '')
+		)
+	end
+
+--[[
+|has teams page={{#resolve_redirect:{{#var:team1|TBD}}}},{{#resolve_redirect:{{#var:team2|TBD}}}}|+sep=,
+]]
+
+	mw.smw.subobject(subObjectTable)
+end
+
+function p.storeGames(match, match2)
+	local games = ''
+	for gameIndex, game2 in ipairs(match2.match2games or {}) do
+		local game = Table.deepCopy(game2)
+		local extradata = Json.parseIfString(game2.extradata)
+
+		game.mode = extradata.maptype
+
+		-- Other stuff
+		game.opponent1 = match.opponent1
+		game.opponent2 = match.opponent2
+		game.opponent1flag = match.opponent1flag
+		game.opponent2flag = match.opponent2flag
+		game.date = match.date
+		local scores = game2.scores or {}
+		if type(scores) == 'string' then
+			scores = Json.parse(scores)
+		end
+		game.opponent1score = scores[1] or 0
+		game.opponent2score = scores[2] or 0
+		local res = mw.ext.LiquipediaDB.lpdb_game(
+			'legacygame_' .. match2.match2id .. gameIndex,
+			game
+		)
+		games = games .. res
+	end
+	return games
+end
+
+function p.convertParameters(match2)
+	local match = Table.deepCopy(match2)
+	for key, _ in pairs(match) do
+		if String.startsWith(key, 'match2') then
+			match[key] = nil
+		end
+	end
+
+	if match.walkover == 'ff' or match.walkover == 'dq' then
+		match.walkover = match.winner
+	elseif match.walkover == 'l' then
+		match.walkover = nil
+	end
+
+	match.staticid = match2.match2id
+
+	-- Handle extradata fields
+	local extradata = Json.parseIfString(match2.extradata)
+	match.extradata = {
+		mvp = extradata.mvp,
+	}
+
+	for index, map in pairs(match2.match2games or {}) do
+		match.extradata['vodgame' .. index] = map.vod
+	end
+	match.extradata.matchsection = extradata.matchsection
+	local bracketData = Json.parseIfString(match2.match2bracketdata)
+	if type(bracketData) == 'table' and bracketData.type == 'bracket' then
+		local headerName
+		if bracketData.header then
+			headerName = (DisplayHelper.expandHeader(bracketData.header) or {})[1]
+		end
+		if String.isEmpty(headerName) then
+			headerName = Variables.varDefault('match_legacy_header_name')
+		end
+		if String.isNotEmpty(headerName) then
+			match.header = headerName
+			Variables.varDefine('match_legacy_header_name', headerName)
+		end
+	end
+
+	-- Handle Opponents
+	local handleOpponent = function (index)
+		local prefix = 'opponent' .. index
+		local opponent = match2.match2opponents[index] or {}
+		local opponentmatch2players = opponent.match2players or {}
+		if opponent.type == 'team' then
+			match.extradata['team' .. index .. 'icon'] = opponent.icon
+			match[prefix] = mw.ext.TeamTemplate.teampage(opponent.template)
+			match[prefix .. 'score'] = (tonumber(opponent.score) or 0) > 0 and opponent.score or 0
+			local opponentplayers = {}
+			for i = 1,10 do
```

Comment:
Up to the top

Reply:
changed it into a while loop instead
btw this is a hard copy from R6, so should be updated there too ;)